### PR TITLE
Enable shellcheck linting of bash files

### LINKED ISSUE #46: Add Shellcheck CI checks
**What would you like to be added**:

Need to add [Shellcheck](https://www.shellcheck.net) CI Checks for all bash scripts in the repo.

**Why is this needed**:

To ensure adherence to a standard for BASH scripts for consistency and best practices. This should help to maintain trusted tooling within the repo.

### PR DESCRIPTION
**Pull request by ilrudie**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind bug
> /kind cleanup
> /kind deprecation

/kind design
> /kind documentation
> /kind feature

**What this PR does / why we need it**:
Enables shellcheck linting of bash files.

Includes logic for ci to exclude problematic bash scripts which were already accpeted from new linting so that ci still runs clean. Will want to correct these under a different issue/pr going forward so that special ci exemptions can be removed. `make lint-shell` can be run locally to determine what needs to be corrected in exempted files. Once corrections are made and local linting of a file comes back clean the exemption can be removed from hack/.shellcheck-selection to bring the script under the new shellcheck linting in ci going forward.

Example of what output looks like when shellcheck is not clean can be found in my fork under some testing pushes [here](https://github.com/ilrudie/magtape/actions/runs/304997979).

**Which issue(s) this PR fixes**:

Fixes #46

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., usage docs, etc.**:

```docs

```

### CODE REVIEW

File: hack/lint-shell.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+RUN_TYPE="${1}"
+SELECTION_FILE="hack/.shellcheck-selection"
+
+################################################################################
+#### Main ######################################################################
+################################################################################
+
+# check to see if being run for ci, if yes exclude legacy scripts from linting
+if [[ "ci" == "${RUN_TYPE}" ]]; then
```
Comment (phenixblue) [inline hack/lint-shell.sh:33]: Just a small nit here. It's usually preferred to compare the variable to the variable to the assumed value.

```suggestion
if [[ "${RUN_TYPE}" == "ci" ]]; then
```

**Review by phenixblue — CHANGES_REQUESTED** [review]
Looks good beyond the one nit.

File: hack/lint-shell.sh (hunk 1 above)
Comment (ilrudie) [reply hack/lint-shell.sh:33]: I do this because it avoids accidental assignment. Let's change all the other stuff to be like this instead.

**Review by phenixblue — APPROVED** [review]
/lgtm

**Pull request merged**